agol-publish/NightStocker.py

```diff
@@ -0,0 +1,520 @@
+            cim.spatialReference = {'wkid':3857}
+            agol_map.setDefinition(cim)
+        else:
+            cim = None
+
+        proj.save()
+
+        item_name = layer_info['title']
+        if not item_name.startswith('Utah'):
+            item_name = f'Utah {item_name}'
+
+        #: Staging
+        print("staging")
+        draft_path = os.path.join(temp_dir, f'{item_name}.sddraft')
+        sd_path = draft_path[:-5]
+        sharing_draft = agol_map.getWebLayerSharingDraft('HOSTING_SERVER',
+                                                         'FEATURE', item_name,
+                                                         [layer])
+        sharing_draft.exportToSDDraft(draft_path)
+        arcpy.server.StageService(draft_path, sd_path)
+
+        end = datetime.datetime.now()
+        print(f'staging time: {end-start}')
+
+    except arcpy.ExecuteError:
+        raise  #: pass error up so that it can be logged and continued
+
+    finally:
+
+        #: Leaving this here for future troubleshooting. Something from
+        #: .addDataFromPath() is not releasing the file handle on the temporary
+        #: fgdb until the parent python process is killed, as if it spawns
+        #: another process that doesn't end until the parent ends. This
+        #: prevents any temp folder cleanup from completing
+        #: (arcpy.Delete_management(), shutil.rmtree(), or 'with
+        #: tempfile.TemporaryDirectory:'). Deleting any and all references to
+        #: the layer and anything in the map doesn't seem to help, nor does
+        #: deleting the feature classes from the temp fgdb prior to trying to
+        #: delete the fgdb itself.
+
+        if layer and not is_table:
+            layer.updateConnectionProperties(os.path.join(temp_dir, 'tempfgdb.gdb'), r'c:\foo\bar.gdb', auto_update_joins_and_relates=False, validate=False)
+
+            agol_map.removeLayer(layer)
+            proj.save()
+
+        # layer = None
+        # cim = None
+        # agol_map = None
+        # proj = None
+        # sharing_draft = None
+
+        # del layer
+        # del cim
+        # del agol_map
+        # del proj
+        # del sharing_draft
+
+        #: Delete feature class
+        # print(f'Deleting {projected_table}...')
+        # arcpy.Delete_management(projected_table)
+        # tempgdb = os.path.join(temp_dir, 'tempfgdb.gdb')
+        # print(f'Deleting {tempgdb}...')
+        # shutil.rmtree(tempgdb)
+
+    return sd_path
+
+
+def get_info(entry, generic_terms_of_use):
+    '''Get the info needed for publishing AGOL item.
+    
+    Parameters:
+    entry: list from CSV: [fully-qualifed FC name, fc title, credit, method]
+    generic_terms_of_use: Standard license info for items that don't have 
+                          license info in their metadata
+    
+    returns: dict of relevant information
+    '''
+    category = entry[0].split('.')[-2].title()
+    credit = entry[2] if entry[2] else 'AGRC'
```

Comment:
I don't think that I've ever seen this syntax before. Cool!

Reply:
It's the python equivalent to the ternary/conditional operator: https://stackoverflow.com/questions/394809/does-python-have-a-ternary-conditional-operator. Nice and succinct.